### ISSUE TITLE
Re-using x25519 Public Key with different peers

### ISSUE THREAD

**Issue by ustulation** [dalek-cryptography/x25519-dalek#46]
Hi, Since contributors here deal a lot with implementation of the cryptographic primitives, I was wondering if some of you also have an opinion on [the question posted here](https://crypto.stackexchange.com/questions/73040/is-using-different-public-keys-for-different-peers-safer-than-reusing-the-public).

To expand that link here:

---

In `X25519` (`ECDH` over `Curve25519`) Peer A and B exchange their Public Keys `PkA and PkB` and then calculate a shared-secret `SecAB` using `combinationOf(SkA, PkB) == combinationOf(SkB, PkA)`.

For forward secrecy, I think it's recommended for A and B to re-negotiate a new shared-secret regularly (thus advertise new Public Keys).

**_Leaving that aside_** is there any other disadvantage if A uses the same Public Key that they have with everyone to get corresponding shared-secret VS if they use new Public Key with each new peer ?

In other words:

 1. Peer A gives `PkA` to all, `B`, `C`, `D` ..., and as usual calculates shared-secret using `PkB`, `PkC`, `PkD` as `SecAB`, `SecAC`, `SecAD` and so on and uses those to encrypt/decrypt messages to the corresponding peers.
 2. Peer A gives new Public Keys to each. So `PkAB` to `B`, `PkAC` to `C`, `PkAD` to `D` and so on. Then, just like before, calculate the shared-secret after using Public Keys of the corresponding peers as `SecAB`, `SecAC`, `SecAD` and so on.

Is `1.` above less secure in context of how `X25519` (and its maths) works than `2.` ? Or is it just wasteful to do `2.` and `1.` is equally good ?

---

Thanks !

**Comment by ustulation** [comment]
Ah the question's answered there now. So it seems it's only from a forward-secrecy point of view that it's recommended to have ephemeral DH keys. Otherwise there's no other reason (like it becomes more susceptible to attack theoretically etc.) . That is all I was after. If anybody has anything more on this pls let me know else feel free to close this. I'll close this issue in a day or two if no one else closes.
